synthetic_test.py

```diff
@@ -0,0 +1,65 @@
+import matplotlib.pyplot as plt
```

Comment:
Is this meant to be something like a unit test?

Reply:
Yeah its a bit of a personal sanity check more than anything but I did envisage eventually adapting it into a unit test.

Reply:
@AlexLipp - If you'd like to make this a unit test we'll probably need to hard-code some numbers and such in it so that we can automatically determine if they're changing. We could also compare output images. A good unit test shouldn't require eyeballing :-)